front/src/components/tracker/trackerHandler.component.tsx

```diff
@@ -20,23 +23,41 @@ const TrackerHandler: FC<TrackerHandlerProps> = ({
   searchObject,
   eventObject,
 }) => {
-  const { trackScreenView, trackAction, trackSiteSearch, trackEvent } =
-    useMatomo();
+  const { trackScreenView, trackAction, trackSiteSearch, trackEvent } = useMatomo();
 
-  const userInfo: TrackerUserInfo = {
-    dimension1: Constants.manifest?.version ?? "", // dimension1 = AppVersion
+  const getUserInfo = async () => {
+    const userSituations = (await StorageUtils.getObjectValue(
+      StorageKeysConstants.userSituationsKey
+    )) as UserSituation[] | null;
+    const gender = (await StorageUtils.getObjectValue(StorageKeysConstants.userGenderKey)) as ProfileGender | null;
+    const currentStepLabel = (await StorageUtils.getStringValue(StorageKeysConstants.currentStepLabelKey)) ?? null;
+    
+    let userSituationLabel = null;
+    if (userSituations) {
+      const userSituation = _.find(userSituations, { isChecked: true });
+      userSituationLabel = userSituation?.label ?? null;
+    }
+    
+    const userInfo: TrackerUserInfo = {
+      dimension1: Constants.manifest?.version ?? "",
```

Comment:
dimension1, 2, 3, 4 : c'est pas très parlant même si tu as mis des commentaires avec la correspondance.

Reply:
Oui c'est pas terrible mais on a pas le choix c'est imposé par l'api Matomo : https://developer.matomo.org/api-reference/tracking-api#optional-user-info (dimension[0-999] — A Custom Dimension value for a specific Custom Dimension ID (requires Matomo 2.15.1 + [Custom Dimensions plugin](https://plugins.matomo.org/CustomDimensions) see the [Custom Dimensions guide](https://matomo.org/docs/custom-dimensions/)). If Custom Dimension ID is 2 use)

Reply:
Ah ok, c'est pour ça.